docs/ccv1_image_security_design.md

```diff
@@ -0,0 +1,345 @@
+                "org.opencontainers.image.enc.keys.provider.attestation-agent": "eyJwcm9..."
+            }
+        }
+    ]
+}
+```
+
+### Image signing
+
+The container image signing standard used by the confidential containers V1 is follow the signing mechanism used in [containers/image](https://github.com/containers/image). At present, the only defined cryptographic signature format is a OpenPGP signature (RFC 4880). The signing process will not modify any content of the container image, including layer, config, manifest, etc. Therefore, the signature will not change the image manifest digest used to identify the image.
+
+The container signature under this standard is essentially a blob, which consists of two parts:
+
+- JSON payload
+
+- Cryptographic signature of JSON payload
+
+The JSON payload contains the manifest digest and identity information of the container image. Examples are as follows:
+
+```json
+{
+    "critical": {
+        "type": "atomic container signature",
+        "image": {
+            "docker-manifest-digest": "sha256:817a12c32a39be..."
+        },
+        "identity": {
+            "docker-reference": "docker.io/library/busybox:latest"
+        }
+    },
+    "optional": {
+        "creator": "some software package v1.0.1-35",
+        "timestamp": 1483228800,
+    }
+}
+```
+
+The process of signing a container image is as follows:
+
+1. Create a signature JSON payload and write the manifest digest together with some otheridentity information of the container image.
+2. Use the private key to sign the JSON payload generated in the previous step, and package the cryptographic signature together with the JSON payload into a signature claim file.
+
+The signature claim file is what we usually call the container image signature file. Its directory structure and file name format are as follows:
+
+```
+.
+└── busybox@sha256=817a12c32a39bbe394944ba49de563e085f1d3c5266eb8e9723256bc4448680e
+    └── signature-1
+```
+
+The meaning of directory name is `image-name@digest-algo=digest-value`, multiple signature files can exist in the directory, which means that the same container image can have multiple signatures and be numbered and stored.
+
+### Image distribution
+
+After the encryption and signing are completed, we obtain two contents: the container image itself and the signature claim file, which are distributed and stored separately.
+
+- container image: pushed to the specified remote registry.
+
+- signature claim file: upload to the specified sigstore server.
+
+## Deployment of protected container image
+
+In the confidential container V1 design, security functions will be performed when pulling the container image, including checking the registry whitelist, verifying the container image signature, and decrypting the container image layer. The high-level process is as follows:
+
+1. Check whether the registry where the image is located is in policy.json's whitelist.
```

Comment:
It would be really nice to change all occurrence of `whitelist` with `allowlist`.

Reply:
Done :)